internal/tools/semconvkit/main.go

```diff
@@ -58,16 +130,156 @@ func render(src, dest string, data *SemanticConventions) error {
 	return nil
 }
 
-func main() {
-	flag.Parse()
+// prevVer returns the previous version of the semantic conventions package.
+// It will first check for hint within root and return that value if found. If
+// not found, it will find all directories in root with a version name and
+// return the version that is less than and closest to the curr version.
+func prevVer(root, curr, hint string) (string, error) {
+	slog.Debug("prevVer", "root", root, "curr", curr, "hint", hint)
+	info, err := os.Stat(root)
+	if err != nil {
+		return "", fmt.Errorf("root directory %q not found: %w", root, err)
+	}
+	if !info.IsDir() {
+		return "", fmt.Errorf("root %q is not a directory", root)
+	}
 
-	if *tag == "" {
-		log.Fatalf("invalid tag: %q", *tag)
+	if hint != "" {
+		sub := filepath.Join(root, hint)
+		slog.Debug("looking for hint", "path", sub)
+		info, err = os.Stat(sub)
+		if err == nil && info.IsDir() {
+			return hint, nil
+		}
 	}
 
-	sc := &SemanticConventions{TagVer: *tag}
+	v, err := semver.NewVersion(curr)
+	if err != nil {
+		return "", fmt.Errorf("invalid current version %q: %w", curr, err)
+	}
 
-	if err := render("templates/*.tmpl", *out, sc); err != nil {
-		log.Fatal(err)
+	entries, err := os.ReadDir(root)
+	if err != nil {
+		return "", fmt.Errorf("error reading root %q: %w", root, err)
+	}
+
+	var prev *semver.Version
+	for _, entry := range entries {
+		slog.Debug("root entry", "name", entry.Name())
+		if !entry.IsDir() {
+			continue
+		}
+
+		ver, err := semver.NewVersion(entry.Name())
+		if err != nil {
+			slog.Debug("not a version dir", "name", entry.Name())
+			// Ignore errors for non-semver directories.
+			continue
+		}
+		slog.Debug("found version dir", "prev", ver)
+		if ver.LessThan(v) && (prev == nil || ver.GreaterThan(prev)) {
+			slog.Debug("new previous version", "version", ver)
+			prev = ver
+		}
+	}
+
+	if prev == nil {
+		return "", errors.New("no previous version found")
+	}
+	return prev.Original(), nil
+}
+
+// SemanticConventions are information about the semantic conventions being
+// generated.
+type SemanticConventions struct {
+	// TagVer is the tagged version (i.e. v.7.0 and not 1.7.0).
+	TagVer string
+}
+
+func (sc SemanticConventions) SemVer() string {
+	return strings.TrimPrefix(*tag, "v")
+}
+
+// Migration contains the details about the migration from the previous
+// semantic conventions to the current one.
+type Migration struct {
+	CurrVer  string
```

Comment:
nit

```suggestion
	CurVer  string
```

I am used to the `cur` abbreviation of "current". Examples:
- Make: https://www.gnu.org/software/make/manual/make.html#index-CURDIR
- Python: https://docs.python.org/3/library/os.html#os.curdir
- VBA: https://learn.microsoft.com/en-us/office/vba/language/reference/user-interface-help/curdir-function